### PR TITLE
dm: update contents about wildcards

### PR DESCRIPTION
**Pull request by Frank945946**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v7.4 (TiDB 7.4 versions)
- [x] v7.3 (TiDB 7.3 versions)
- [x] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: dm/dm-block-allow-table-lists.md [hunk 1]
```diff
@@ -33,10 +33,10 @@ block-allow-list:             # 如果 DM 版本早于 v2.0.0-beta.2 则使用 b
       tbl-name: "log"
 ```
 
-在简单任务场景下，推荐使用通配符匹配库表名，但需注意以下版本差异：
+黑白名单支持通配符和正则表达式来匹配，在简单任务场景下，推荐使用通配符匹配库表名：
 
-+ 对于 v1.0.5 版及后续版本，黑白名单支持[通配符匹配](https://en.wikipedia.org/wiki/Glob_(programming)#Syntax)。但注意所有版本中通配符匹配中的 `*` 符号 **只能有一个，且必须在末尾**。
-+ 对于 v1.0.5 以前的版本，黑白名单仅支持正则表达式。
++ 通配符支持`*`、`?` 以及 `[]`。通配符匹配中的 `*` 符号 ** 只能有一个，且必须在末尾，如 `tbl-name: "test*"` 表示 `test` 开头的表 **。详情请参考[通配符匹配](https://en.wikipedia.org/wiki/Glob_(programming)#Syntax)。
```
Comment (hfxsd) [inline dm/dm-block-allow-table-lists.md:38]:
```suggestion
+ 通配符支持`*`、`?` 以及 `[]`。通配符匹配中的 `*` 符号只能有一个，且必须在末尾，如用 `tbl-name: "test*"` 表示以 `test` 开头的表。详情请参考[通配符匹配](https://en.wikipedia.org/wiki/Glob_(programming)#Syntax)。
```

File: dm/dm-binlog-event-filter.md [hunk 2]
```diff
@@ -23,10 +23,10 @@ filters:
 
 从 DM v2.0.2 开始，你也可以在上游数据库配置文件中配置 Binlog 事件过滤。见[上游数据库配置文件介绍](/dm/dm-source-configuration-file.md)。
 
-在简单任务场景下，推荐使用通配符匹配库表名，但需注意以下版本差异：
+注意以下事项：
 
-+ 对于 v1.0.5 版及后续版本，Binlog 事件过滤支持[通配符匹配](https://en.wikipedia.org/wiki/Glob_(programming)#Syntax)。但注意所有版本中通配符匹配中的 `*` 符号 **只能有一个，且必须在末尾**。
-+ 对于 v1.0.5 以前的版本，Binlog Event Filter 支持通配符，但不支持 `[...]` 与 `[!...]` 表达式。
++ `schema-pattern` 和 `​table-pattern` 仅支持通配符，且通配符仅支持 `*`、`?` 和 `[]`。通配符匹配中的 `*` 符号只能有一个，且必须在末尾。例如 `table-pattern: "t_*"` 表示以 `t_` 开头的表。详情请参考[通配符匹配](https://en.wikipedia.org/wiki/Glob_(programming)#Syntax)。
++ `sql-pattern` 仅支持正则表达式。
```
Comment (hfxsd) [inline dm/dm-binlog-event-filter.md:29]: 这里需要加上这句话吗？正则表达式必须以 `~` 符号开头。

Comment (lance6716) [reply dm/dm-binlog-event-filter.md:29]: 不需要的

**Review by lance6716 — APPROVED** [review]

File: dm/dm-table-routing.md [hunk 3]
```diff
@@ -38,10 +38,12 @@ routes:
     target-schema: "test"
 ```
 
-在简单任务场景下，推荐使用通配符匹配库表名，但需注意以下版本差异：
+支持正则表达式和通配符来匹配库表名，在简单任务场景下，推荐使用通配符匹配库表名，但需注意以下几点：
+
++ 通配符仅支持 `*`、`?` 以及 `[]`。符号 `*` 只能有一个，且必须在末尾。例如用 `tbl-name:"test*"` 表示 `test` 开头的表。详情请参考[通配符匹配](https://en.wikipedia.org/wiki/Glob_(programming)#Syntax)。
```
Comment (hfxsd) [inline dm/dm-table-routing.md:43]: 上面示例中没有 `tbl-name:"test*"`，需要改一下

Comment (hfxsd) [inline dm/dm-table-routing.md:43]:
```suggestion
+ 通配符仅支持 `*`、`?` 以及 `[]`。符号 `*` 只能有一个，且必须在末尾。例如用 `tbl-name: "test*"` 表示 `test` 开头的表。详情请参考[通配符匹配](https://en.wikipedia.org/wiki/Glob_(programming)#Syntax)。
```

File: dm/dm-binlog-event-filter.md [hunk 4]
```diff
@@ -23,10 +23,10 @@ filters:
 
 从 DM v2.0.2 开始，你也可以在上游数据库配置文件中配置 Binlog 事件过滤。见[上游数据库配置文件介绍](/dm/dm-source-configuration-file.md)。
 
-在简单任务场景下，推荐使用通配符匹配库表名，但需注意以下版本差异：
+注意以下事项：
```
Comment (hfxsd) [inline dm/dm-binlog-event-filter.md:26]:
```suggestion
使用通配符匹配库表名时，注意以下事项：
```

File: dm/dm-block-allow-table-lists.md [hunk 5]
```diff
@@ -33,10 +33,10 @@ block-allow-list:             # 如果 DM 版本早于 v2.0.0-beta.2 则使用 b
       tbl-name: "log"
 ```
 
-在简单任务场景下，推荐使用通配符匹配库表名，但需注意以下版本差异：
+黑白名单支持通配符和正则表达式来匹配，在简单任务场景下，推荐使用通配符匹配库表名：
 
-+ 对于 v1.0.5 版及后续版本，黑白名单支持[通配符匹配](https://en.wikipedia.org/wiki/Glob_(programming)#Syntax)。但注意所有版本中通配符匹配中的 `*` 符号 **只能有一个，且必须在末尾**。
-+ 对于 v1.0.5 以前的版本，黑白名单仅支持正则表达式。
++ 通配符支持`*`、`?` 以及 `[]`。注意通配符匹配中的 `*` 符号只能有一个，且必须在末尾，例如用 `tbl-name: "test*"` 表示以 `test` 开头的表。详情请参考[通配符匹配](https://en.wikipedia.org/wiki/Glob_(programming)#Syntax)。
```
Comment (hfxsd) [inline dm/dm-block-allow-table-lists.md:38]:
```suggestion
+ 通配符支持`*`、`?` 以及 `[]`。注意通配符匹配中的 `*` 符号只能有一个，且必须在末尾，例如用 `tbl-name: "t*"` 表示以 `t` 开头的表。详情请参考[通配符匹配](https://en.wikipedia.org/wiki/Glob_(programming)#Syntax)。
```

**Review by Frank945946 — COMMENTED** [review]
LGTM

**Review by hfxsd — APPROVED** [review]

File: dm/dm-table-routing.md [hunk 6]
```diff
@@ -38,10 +38,11 @@ routes:
     target-schema: "test"
 ```
 
-在简单任务场景下，推荐使用通配符匹配库表名，但需注意以下版本差异：
+支持正则表达式和通配符来匹配库表名，在简单任务场景下，推荐使用通配符匹配库表名，但需注意以下几点：
 
-+ 对于 v1.0.5 版及后续版本，表路由支持[通配符匹配](https://en.wikipedia.org/wiki/Glob_(programming)#Syntax)。但注意所有版本中通配符匹配中的 `*` 符号 **只能有一个，且必须在末尾**。
-+ 对于 v1.0.5 以前的版本，表路由支持通配符，但不支持 `[...]` 与 `[!...]` 表达式。
++ 通配符仅支持 `*`、`?` 以及 `[]`。符号 `*` 只能有一个，且必须在末尾。例如用 `table-pattern: "t_*"` 表示 `t_` 开头的表。详情请参考[通配符匹配](https://en.wikipedia.org/wiki/Glob_(programming)#Syntax)。
++ `table-regexp`、`schema-regexp` 和 `source-regexp` 仅支持配置正则表达式，且不能以 `~` 符号开头。
```
Comment (hfxsd) [inline dm/dm-table-routing.md:44]:
```suggestion
+ `table-regexp`、`schema-regexp` 和 `source-regexp` 仅支持配置正则表达式，但不能以 `~` 符号开头。
```

File: dm/dm-block-allow-table-lists.md [hunk 7]
```diff
@@ -27,16 +27,16 @@ block-allow-list:             # 如果 DM 版本早于 v2.0.0-beta.2 则使用 b
...
     - db-name: "test"
       tbl-name: "log"
 ```
 
-在简单任务场景下，推荐使用通配符匹配库表名，但需注意以下版本差异：
+黑白名单支持通配符和正则表达式来匹配，在简单任务场景下，推荐使用通配符匹配库表名：
 
-+ 对于 v1.0.5 版及后续版本，黑白名单支持[通配符匹配](https://en.wikipedia.org/wiki/Glob_(programming)#Syntax)。但注意所有版本中通配符匹配中的 `*` 符号 **只能有一个，且必须在末尾**。
-+ 对于 v1.0.5 以前的版本，黑白名单仅支持正则表达式。
++ 通配符支持`*`、`?` 以及 `[]`。注意通配符匹配中的 `*` 符号只能有一个，且必须在末尾，例如 `tbl-name: "t*"` 中的 `"t*"` 表示以 `t` 开头的表。详情请参考[通配符匹配](https://en.wikipedia.org/wiki/Glob_(programming)#Syntax)。
```
Comment (hfxsd) [inline dm/dm-block-allow-table-lists.md:38]:
```suggestion
+ 支持的通配符包括 `*`、`?` 以及 `[]`。注意通配符匹配中的 `*` 符号只能有一个，且必须在末尾，例如 `tbl-name: "t*"` 中的 `"t*"` 表示以 `t` 开头的表。详情请参考[通配符匹配](https://en.wikipedia.org/wiki/Glob_(programming)#Syntax)。
```

File: dm/dm-block-allow-table-lists.md [hunk 8]
```diff
@@ -27,16 +27,16 @@ block-allow-list:             # 如果 DM 版本早于 v2.0.0-beta.2 则使用 b
...
++ 正则表达式需要以 `~` 字符开头。
```
Comment (hfxsd) [inline dm/dm-block-allow-table-lists.md:39]:
```suggestion
+ 正则表达式必须以 `~` 字符开头。 
```

File: dm/dm-table-routing.md [hunk 9]
```diff
@@ -38,10 +38,11 @@ routes:
     target-schema: "test"
 ```
 
-在简单任务场景下，推荐使用通配符匹配库表名，但需注意以下版本差异：
+支持正则表达式和通配符来匹配库表名，在简单任务场景下，推荐使用通配符匹配库表名，但需注意以下几点：
 
-+ 对于 v1.0.5 版及后续版本，表路由支持[通配符匹配](https://en.wikipedia.org/wiki/Glob_(programming)#Syntax)。但注意所有版本中通配符匹配中的 `*` 符号 **只能有一个，且必须在末尾**。
-+ 对于 v1.0.5 以前的版本，表路由支持通配符，但不支持 `[...]` 与 `[!...]` 表达式。
++ 通配符仅支持 `*`、`?` 以及 `[]`。`*` 符号只能有一个，且必须在末尾。例如用 `table-pattern: "t_*"` 中的 `"t_*"` 表示 `t_` 开头的表。详情请参考[通配符匹配](https://en.wikipedia.org/wiki/Glob_(programming)#Syntax)。
```
Comment (hfxsd) [inline dm/dm-table-routing.md:43]:
```suggestion
+ 支持的通配符包括 `*`、`?` 以及 `[]`。注意通配符匹配中的 `*` 符号只能有一个，且必须在末尾。例如用 `table-pattern: "t_*"` 中的 `"t_*"` 表示 `t_` 开头的表。详情请参考[通配符匹配](https://en.wikipedia.org/wiki/Glob_(programming)#Syntax)。
```

File: dm/dm-binlog-event-filter.md [hunk 10]
```diff
@@ -23,10 +23,10 @@ filters:
 
 从 DM v2.0.2 开始，你也可以在上游数据库配置文件中配置 Binlog 事件过滤。见[上游数据库配置文件介绍](/dm/dm-source-configuration-file.md)。
 
-在简单任务场景下，推荐使用通配符匹配库表名，但需注意以下版本差异：
+使用通配符匹配库表名时，注意以下事项：
 
-+ 对于 v1.0.5 版及后续版本，Binlog 事件过滤支持[通配符匹配](https://en.wikipedia.org/wiki/Glob_(programming)#Syntax)。但注意所有版本中通配符匹配中的 `*` 符号 **只能有一个，且必须在末尾**。
-+ 对于 v1.0.5 以前的版本，Binlog Event Filter 支持通配符，但不支持 `[...]` 与 `[!...]` 表达式。
++ `schema-pattern` 和 `​table-pattern` 仅支持通配符，且通配符仅支持 `*`、`?` 和 `[]`。通配符匹配中的 `*` 符号只能有一个，且必须在末尾。例如 `table-pattern: "t_*"` 中的 `"t_*"` 表示以 `t_` 开头的表。详情请参考[通配符匹配](https://en.wikipedia.org/wiki/Glob_(programming)#Syntax)。
```
Comment (hfxsd) [inline dm/dm-binlog-event-filter.md:28]:
```suggestion
+ `schema-pattern` 和 `​table-pattern` 仅支持通配符，支持的通配符包括 `*`、`?` 和 `[]`。注意通配符匹配中的 `*` 符号只能有一个，且必须在末尾。例如 `table-pattern: "t_*"` 中的 `"t_*"` 表示以 `t_` 开头的表。详情请参考[通配符匹配](https://en.wikipedia.org/wiki/Glob_(programming)#Syntax)。
```

**Review by ran-huang — APPROVED** [review]

**Pull request merged**